kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,288 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.freedesktop.dbus.errors.NotSupported;
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties;
+	Optional<?> optResult;
+	String stringResult;
+	List<String> stringListResult;
+	Map<String, Object> resultMap;
+
+	Boolean hasNullPointExceptionBeenThrown = false;
+	Boolean hasNoSuchElementExceptionBeenThrown = false;
+
+	@Test
+	public void shouldFailWhenNull() {
+		givenNetworkPropertiesBuiltWith(null);
+		givenNetworkPropsIsCreated();
+		thenAnNullPointerExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnPropertiesWithGetProperties() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetPropertiesIsCalled();
+		thenResultEquals(Collections.singletonMap("testKey1", "testString1"));
```

Comment:
You should also check that no exception occurred for each tests where you expect it, right?

```suggestion
		thenNoExceptionOccurred();
		thenResultEquals(Collections.singletonMap("testKey1", "testString1"));
```